cmd/cluster/logs/vault.go

```diff
@@ -0,0 +1,53 @@
+package logs
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+)
+
+type VaultResponse struct {
+	Data struct {
+		Data map[string]interface{} `json:"data"`
+	} `json:"data"`
+}
+
+func getSecretFromVault() (string, error) {
+	err := os.Setenv("VAULT_ADDR", vaultAddr)
+	if err != nil {
+		fmt.Printf("Error setting environment variable: %v\n", err)
+		return "", err
+	}
+	loginCommand := exec.Command("vault", "login", "-method=oidc", "-no-print")
+	loginCommand.Stdout = nil
+	loginCommand.Stderr = nil
+	if err := loginCommand.Run(); err != nil {
+		fmt.Println("Error running 'vault login':", err)
+		return "", nil
```

Comment:
**chore (repeating):** Return the error

Reply:
This comment is still open. Is there a reason we're not returning them to let them be returned in the main logic?